src/box/memtx_engine.cc

```diff
@@ -900,8 +971,18 @@ is_tuple_temporary(const char *data, uint32_t space_id,
 	}
 }
 
+static int
+checkpoint_dump_sort_data(
+	struct index_read_view *rv, ssize_t tuple_count,
+	struct memtx_sort_data_writer *writer, bool *have_more)
+{
+	assert(rv->def->type == TREE);
+	return memtx_tree_index_read_view_dump_sort_data(rv, tuple_count,
+							 writer, have_more);
+}
+
 static struct checkpoint *
-checkpoint_new(const char *snap_dirname, uint64_t snap_io_rate_limit)
+checkpoint_new(struct memtx_engine *memtx)
```

Comment:
Nit: passing a third argument `sort_data_enabled` here seems nicer to me.

But if you want to pass `memtx` here - probably, it would be nice to do this refactoring in a separate commit.

Reply:
Regarding a separate commit, see the [comment above](https://github.com/tarantool/tarantool/pull/11133#discussion_r2094546662).

Reply:
Changed the argument in a separated commit.